spec/features/photos_feature_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+feature 'photos' do
+  
+  before do
+    @user = User.create email: 'natso@gmail.com', password: '12345678', password_confirmation: '12345678'
+    login_as @user
+  end
+
+  context 'no photos have been added' do
+    scenario 'prompts user to upload a photo' do
+      visit '/photos'
+      expect(page).to have_content 'No photos yet'
+      expect(page).to have_link 'Upload a photo'
+    end
+  end
+
+  context 'adding a new photo' do
+  	scenario 'allows user to upload a photo' do
+      visit '/'
+      click_link 'Upload a photo'
+      fill_in 'Description', with: 'Instrumental'
+      attach_file 'Image', 'spec/fixtures/files/gramophone.png'
+      click_button 'Create Photo'
+      expect(page).to have_content 'Instrumental'
+      expect(page).not_to have_content 'No photos yet'
+      expect(page).to have_xpath("//img[contains(@src, 'thumb/gramophone.png')]")
+    end
+
+    scenario 'photo must be provided' do
+      visit '/'
+      click_link 'Upload a photo'
+      fill_in 'Description', with: 'Instrumental'
+      click_button 'Create Photo'
+      expect(page).not_to have_content 'Instrumental'
+      expect(page).to have_content('Please choose an image')
+    end
+
+    scenario 'denied if not logged in' do
+      logout
+      visit '/'
+      expect(page).not_to have_selector(:link_or_button, 'Upload a photo')
+    end
+  end
+
+  context 'viewing photos' do
+  	let!(:gram) { Photo.create(description:'nice', image_file_name:'spec/fixtures/files/gramophone.png') }
```

Comment:
Line is too long. [105/80]<br>Use 2 (not 1) spaces for indentation.<br>Space missing after colon.<br>Tab detected.